src/haversine.jl

```diff
@@ -0,0 +1,42 @@
+"""
+    Haversine(radius)
+
+The haversine distance between two locations on a sphere of given `radius`.
+
+Locations are described with longitude and latitude in degrees and
+the radius of the Earth is used by default (≈ 6371km). The computed
+distance has the same units as that of the radius.
+
+### Notes
+
+The haversine formula is widely used to approximate the geodesic distance
+between two points at the surface of the Earth. The error from approximating
+the Earth as a sphere is typically negligible for most applications. It is
+no more than 0.3%.
```

Comment:
This is an interesting anecdote but it seems too specific to have in a package as general as Distances.

Reply:
I can remove the comment @ararslan if you like, should I?

Reply:
I think it would be good to remove it